beacon_node/http_api/src/lib.rs

```diff
@@ -1933,6 +1933,51 @@ pub fn serve<T: BeaconChainTypes>(
             })
         });
 
+    /*
+     * builder/states
+     */
+
+    let builder_states_path = eth_v1
+        .and(warp::path("builder"))
+        .and(warp::path("states"))
+        .and(chain_filter.clone());
+
+    // GET builder/states/{state_id}/expected_withdrawals
```

Comment:
This endpoint also accepts an optional `proposal_slot` query string.

https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Builder/getNextWithdrawals